vlog4j-client/src/main/java/org/vlog4j/client/picocli/VLog4jClientMaterialize.java

```diff
@@ -0,0 +1,187 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.semanticweb.vlog4j.core.model.api.PositiveLiteral;
+import org.semanticweb.vlog4j.core.reasoner.Algorithm;
+import org.semanticweb.vlog4j.core.reasoner.KnowledgeBase;
+import org.semanticweb.vlog4j.core.reasoner.LogLevel;
+import org.semanticweb.vlog4j.core.reasoner.Reasoner;
+import org.semanticweb.vlog4j.core.reasoner.implementation.VLogReasoner;
+import org.semanticweb.vlog4j.examples.ExamplesUtils;
+import org.semanticweb.vlog4j.parser.ParsingException;
+import org.semanticweb.vlog4j.parser.RuleParser;
+
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(name = "materialize", description = "Execute the chase and store the literal's extensions")
+public class VLog4jClientMaterialize implements Runnable {
+
+	// TODO add link to rls syntax
+	@Option(names = "--rule-file", description = "Rule file(s) in rls syntax", required = true)
+	private List<String> ruleFiles = new ArrayList<>();
+
+//  TODO
+//	Support graal rule files
+//	@Option(names = "--graal-rule-file", description = "Rule file(s) in graal syntax", required = true)
+//	private List<String> graalRuleFiles = new ArrayList<>();
+
+	@Option(names = "--log-level", description = "Log level of VLog (c++ library). One of: DEBUG, INFO, WARNING (default), ERROR.", required = false)
+	private LogLevel logLevel = LogLevel.WARNING;
+
+	@Option(names = "--log-file", description = "Log file of VLog (c++ library). VLog will log to the default system output by default", required = false)
+	private String logFile;
+
+	@Option(names = "--chase-algorithm", description = "Chase algorithm. RESTRICTED_CHASE (default) or SKOLEM_CHASE.", required = false)
+	private Algorithm chaseAlgorithm = Algorithm.RESTRICTED_CHASE;
+
+	@Option(names = "--timeout", description = "Timeout in seconds. Infinite by default", required = false)
+	private int timeout = 0;
+
+	@Option(names = "--query", description = "Positive not-ground Literals to query after materialization in rls syntax. Vlog4jClient will print the size of its extension", required = true)
+	private List<String> queryStrings = new ArrayList<>();
+
+	@ArgGroup(exclusive = false)
+	private PrintQueryResults printQueryResults = new PrintQueryResults();
+
+	@ArgGroup(exclusive = false)
+	private SaveQueryResults saveQueryResults = new SaveQueryResults();
+
+	// TODO save model
+	// @ArgGroup(exclusive = false)
+	// private SaveModel saveModel = new SaveModel();
+
+	@Override
+	public void run() {
+		ExamplesUtils.configureLogging();
+
+		if (!printQueryResults.isConfigOk()) {
+			printQueryResults.printErrorAndExit();
+		}
```

Comment:
you can extract these lines into a method
`printQueryResults.validate()`

Reply:
Thank you. I've created the method and updated the tests accordingly